mailpoet/lib/API/JSON/ResponseBuilders/DynamicSegmentsResponseBuilder.php

```diff
@@ -82,13 +82,19 @@ private function addMissingPluginProperties(SegmentEntity $segment, array $data)
     if ($missingPlugins) {
       $missingPlugin = reset($missingPlugins);
       $data['is_plugin_missing'] = true;
-      $data['missing_plugin_message'] = $this->segmentDependencyValidator->getCustomErrorMessage($missingPlugin)
-        ?:
-        sprintf(
-          // translators: %s is the name of the missing plugin.
-          __('Activate the %s plugin to see the number of subscribers and enable the editing of this segment.', 'mailpoet'),
-          $missingPlugin
-        );
+
+      $missingPluginMessage = $this->segmentDependencyValidator->getCustomErrorMessage($missingPlugin);
```

Comment:
I wasn't sure how to tell TypeScript that `missing_plugin_message` (https://github.com/mailpoet/mailpoet/pull/4999/files#diff-6d0f885b48ca9cb6143374ead90096ca7ed547142b27db8480cccd6607746e7bR24) can be a string or an object and that is why I changed the structure on the PHP side.

Reply:
In any case, it's a nice improvement to unify the response format :+1: